L1Trigger/L1CaloTrigger/plugins/Phase2L1CaloPFClusterEmulator.h

```diff
@@ -0,0 +1,278 @@
+    r.s7.t[i].phi = i;
+    r.s8.t[i].et = temp[8][i];
+    r.s8.t[i].eta = 8;
+    r.s8.t[i].phi = i;
+    r.s9.t[i].et = temp[9][i];
+    r.s9.t[i].eta = 9;
+    r.s9.t[i].phi = i;
+    r.s10.t[i].et = temp[10][i];
+    r.s10.t[i].eta = 10;
+    r.s10.t[i].phi = i;
+    r.s11.t[i].et = temp[11][i];
+    r.s11.t[i].eta = 11;
+    r.s11.t[i].phi = i;
+    r.s12.t[i].et = temp[12][i];
+    r.s12.t[i].eta = 12;
+    r.s12.t[i].phi = i;
+    r.s13.t[i].et = temp[13][i];
+    r.s13.t[i].eta = 13;
+    r.s13.t[i].phi = i;
+    r.s14.t[i].et = temp[14][i];
+    r.s14.t[i].eta = 14;
+    r.s14.t[i].phi = i;
+    r.s15.t[i].et = temp[15][i];
+    r.s15.t[i].eta = 15;
+    r.s15.t[i].phi = i;
+    r.s16.t[i].et = temp[16][i];
+    r.s16.t[i].eta = 16;
+    r.s16.t[i].phi = i;
+    r.s17.t[i].et = temp[17][i];
+    r.s17.t[i].eta = 17;
+    r.s17.t[i].phi = i;
+    r.s18.t[i].et = temp[18][i];
+    r.s18.t[i].eta = 18;
+    r.s18.t[i].phi = i;
+    r.s19.t[i].et = temp[19][i];
+    r.s19.t[i].eta = 19;
+    r.s19.t[i].phi = i;
+    r.s20.t[i].et = temp[20][i];
+    r.s20.t[i].eta = 20;
+    r.s20.t[i].phi = i;
+
+  }
+  return r;
+}
+
+float getEt(float temp[21][8], int eta, int phi){
+  float et_sumEta[3];
+
+  for(int i=0; i<19; i++){
+    for(int j=0; j<6; j++){
+      if (i+1 == eta && j+1 == phi){
+        for(int k=0; k<3; k++){
+          et_sumEta[k] = temp[i+k][j] + temp[i+k][j+1] + temp[i+k][j+2];
+        }
+      }
+    }
+  }
+
+  float pfcluster_et = et_sumEta[0] + et_sumEta[1] + et_sumEta[2];
+
+  return pfcluster_et;
+}
+
+void RemoveTmp(float temp[21][8], int eta, int phi){
+ 
+  for(int i=0; i<21; i++){
+    if(i+1 >= eta && i <= eta+1) {
+      for(int j=0; j<8; j++){
+        if(j+1 >= phi && j <= phi+1) {
+          temp[i][j] = 0;
+        }
+      }
+    }
+  }
+  return;
+}
+
+GCTpfcluster_t recoPfcluster(float temporary[21][8], int etaoffset, int phioffset){
+  GCTpfcluster_t pfclusterReturn;
+
+  region_t region;
+
+  region = initStructure(temporary);
+
+  GCTint_t regionMax = getPeakPosition(region);  
+
+  float pfcluster_et = getEt(temporary, regionMax.eta, regionMax.phi);
+
+  RemoveTmp(temporary, regionMax.eta, regionMax.phi);
+
+  if(!(regionMax.eta >= 2 && regionMax.eta <= 18 && regionMax.phi >= 2 && regionMax.phi <= 5)) pfcluster_et = 0;
+
+  pfclusterReturn.et  = pfcluster_et;
+  pfclusterReturn.eta = regionMax.eta - 2 + etaoffset;
+  pfclusterReturn.phi = regionMax.phi - 2 + phioffset;
+
+  return pfclusterReturn ;
+}
+
+GCTPfcluster_t pfcluster(float temporary[21][8], int etaoffset, int phioffset){
+  GCTpfcluster_t pfcluster[8];
+
+  for(int i=0; i<8; i++){
+    pfcluster[i] = recoPfcluster(temporary, etaoffset, phioffset);
+  }
+
+  GCTPfcluster_t GCTPfclusters;
+
+  for(int i=0; i<8; i++){
+    GCTPfclusters.GCTpfclusters[i].et = pfcluster[i].et;
+    GCTPfclusters.GCTpfclusters[i].eta = pfcluster[i].eta;
+    GCTPfclusters.GCTpfclusters[i].phi = pfcluster[i].phi;
+  }
+
+  return GCTPfclusters;
+}
+
+#endif
```

Comment:
These functions are also all at global scope, which is also very likely not allowed. It would be better to associate these with a namespace